### PR TITLE
Create CADAUDIT_check.py

### PR DESCRIPTION
**Pull request by nasmovk**
My proposal for a comprehensive audit of cad instances. See details described in the ModelChecker class comments. It would replace the old LISTDWGs by F. Beaupere, from the Project panel, link tools. Bonus: It also extends the basic functionality of RPW by adding radio buttons and a cancel button.

### CODE REVIEW

**Comment by jmcouffin** [comment]
> It also extends the basic functionality of RPW by adding radio buttons and a cancel button.

would be cool if you could add it to the rpw module and refer to it

**Comment by dosymep** [comment]
please add screenshots or any results how to work your script

**Comment by nasmovk** [comment]
@dosymep Sure.
The check starts like this: 
image
Then you get a choice of run mode (using RPW with added radio buttons)
image
The output is in a table with clickable links back to the elements and a legend of explanations of the checks made and why.
image
There is some feedback on it in the forum:
https://discourse.pyrevitlabs.io/t/hackathon-preflight-checks-maker-fair/7702/8

**Comment by nasmovk** [comment]
> > It also extends the basic functionality of RPW by adding radio buttons and a cancel button.
> 
> would be cool if you could add it to the rpw module and refer to it

Yes. I thought about doing that, but I didn't think it was allowed to change the built-in libraries. How would I do it? Make a PR on the RPW files in question?

**Comment by jmcouffin** [comment]
@nasmovk yes, that and writing and adding it to the rpw test in the dev extension.

The rpw lib has been maintained mostly here by pyrevit users

File: extensions/pyRevitTools.extension/checks/CADAUDIT_check.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    #cadinstances = DB.FilteredElementCollector(doc).OfClass(DB.ImportInstance).WhereElementIsNotElementType().ToElements()
+    if active_view_only:
+        cadinstances = DB.FilteredElementCollector(doc, ac_view.Id).OfClass(DB.ImportInstance).WhereElementIsNotElementType().ToElements()
+    else:
+        cadinstances = DB.FilteredElementCollector(doc).OfClass(DB.ImportInstance).WhereElementIsNotElementType().ToElements()
+    
+    if len(cadinstances) >0:
+        return cadinstances, len(cadinstances)
+    else:
+        print("No CAD instances found in the {}.".format("active view" if active_view_only else "model"))
```
Comment (dosymep) [inline extensions/pyRevitTools.extension/checks/CADAUDIT_check.py]: perhaps it is better to use [forms.alert(exitscript=True)](https://docs.pyrevitlabs.io/reference/pyrevit/forms/#pyrevit.forms.alert) here

image

File: extensions/pyRevitTools.extension/checks/CADAUDIT_check.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+    return user_input_dict
+
+def get_load_stat(cad, is_link):
+    """ Loaded status from the import instance's CADLinkType """
+    cad_type = doc.GetElement(cad.GetTypeId()) # Retreive the type from the instance
+    
+    if is_link:
+        exfs = cad_type.GetExternalFileReference()
+        status = exfs.GetLinkedFileStatus().ToString()
```
Comment (dosymep) [inline extensions/pyRevitTools.extension/checks/CADAUDIT_check.py]: to compare enumerations, it is better not to convert them to string `if status == LinkedFileStatus.Loaded`, [see docs](https://www.revitapidocs.com/2022/410b48f8-2a32-3d54-492c-c9a9fe4030ca.htm)

**Review by dosymep — CHANGES_REQUESTED** [review]

File: extensions/pyRevitTools.extension/checks/CADAUDIT_check.py (hunk 2 above)
Comment (nasmovk) [reply extensions/pyRevitTools.extension/checks/CADAUDIT_check.py]: I was getting the string value to be able to return it as a message

File: extensions/pyRevitTools.extension/checks/CADAUDIT_check.py (hunk 1 above)
Comment (jmcouffin) [reply extensions/pyRevitTools.extension/checks/CADAUDIT_check.py]: fixed by commit 9fd9c57950dc859e12dd2c45bf724827d94fc838

File: extensions/pyRevitTools.extension/checks/CADAUDIT_check.py (hunk 2 above)
Comment (jmcouffin) [reply extensions/pyRevitTools.extension/checks/CADAUDIT_check.py]: agree with @dosymep, not the most elegant way/
will merge anyway for now

**Pull request merged**